lib/block-supports/duotone.php

```diff
@@ -459,20 +459,30 @@ function gutenberg_render_duotone_support( $block_content, $block ) {
 	wp_add_inline_style( $filter_id, $filter_style );
 	wp_enqueue_style( $filter_id );
 
-	// Render any custom filter the user may have added.
 	add_action(
-		// There are a couple of known rendering quirks in Safari.
-		// 1. Filters won't render at all when the SVG is in the head of
-		// the document.
-		// 2. Filters display incorrectly when the SVG is defined after
-		// where the filter is used in the document, so the footer does
-		// not work.
-		'wp_body_open',
+		is_admin() ? 'admin_footer' : 'wp_footer',
 		function () use ( $filter_svg ) {
 			echo $filter_svg;
 		}
 	);
 
+	// Safari renders elements incorrectly on first paint when the SVG
+	// filter comes after the content that it is filtering, so we force
+	// a repaint by resetting the display style which solves the issue.
+	global $is_safari;
+	if ( $is_safari ) {
+		wp_register_script( $filter_id, false, array(), null, true );
+		wp_enqueue_script( $filter_id );
+		wp_add_inline_script(
+			$filter_id,
+			sprintf(
+				'( function() { var el = document.querySelector( %s ); var display = el.style.display; el.style.display = "none"; setTimeout( function() { el.style.display = display; } ); } )();',
```

Comment:
Is the setTimeout necessary here?

Reply:
Simply setting and resetting the display property doesn't do anything. We're not dealing with layout like the blog post I shared. And all other "repaint this element" tricks I found used `setTimeout`. So I assumed it was necessary. However, it seems the `offsetHeight` trick works fine to flush the style changes without `setTimeout`.

I dug into the WebKit source to get to the bottom of why this works. I found that some of the element accessors, including [`Element::offsetHeight`](https://github.com/WebKit/WebKit/blob/4121ca9891ada0a3deaba8ee70e3a575407f508d/Source/WebCore/dom/Element.cpp#L1296-L1304), call [`Document::updateLayoutIgnorePendingStylesheets`](https://github.com/WebKit/WebKit/blob/4121ca9891ada0a3deaba8ee70e3a575407f508d/Source/WebCore/dom/Document.cpp#L2237-L2254) which, not only does a reflow, but also immediately recalculates styles with [`Document::updateStyleIfNeeded`](https://github.com/WebKit/WebKit/blob/4121ca9891ada0a3deaba8ee70e3a575407f508d/Source/WebCore/dom/Document.cpp#L2185-L2200). I couldn't find any other easy ways to flush the style changes without reflow, but I don't see a flash anymore with the `offsetHeight` trick, at least on my MacBook Pro M1 Max. 🤷